lib/canvas_exporting/exporter.rb

```diff
@@ -13,14 +13,28 @@ def export
       type = params[:type] || 'image/png'
       extension = MIME::Types[type].first.extensions.first
 
-      @output_tmp_file = Tempfile.new(['output', ".#{extension}"], tmp_dir)
-      filename = params[:filename] || 'Chart'
+      if params[:outputpath] == nil
+        output_path = tmp_dir
+      else
+        output_path = params[:outputpath]
+      end
+
+      output_path = output_path + '/' if output_path[-1] != '/'
+
+      if params[:filename] == nil
+        filename = 'Chart.' + extension
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.